controllers/apply/form-router-next/lib/file-uploads.js

```diff
@@ -109,6 +109,31 @@ function uploadFile({ formId, applicationId, fileMetadata }) {
     });
 }
 
+function getObjectTagging(key) {
+    return s3
+        .getObjectTagging({
+            Bucket: S3_UPLOAD_BUCKET,
+            Key: key
+        })
+        .promise();
+}
+
+function checkAntiVirus({ formId, applicationId, filename }) {
```

Comment:
You shouldn't need to construct a `new Promise` here or use `async` as you are already returning a promise. Could try something like:

```suggestion
function checkAntiVirus({ formId, applicationId, filename }) {
    const keyName = [formId, applicationId, filename].join('/');
    return s3
        .getObjectTagging({
            Bucket: S3_UPLOAD_BUCKET,
            Key: keyName
        })
        .promise()
        .then(function(tags) {
            const tagSet = get(tags, 'TagSet', []);
            const avStatus = tagSet.find(t => t.Key === 'av-status');
            if (avStatus && get(avStatus, 'Value') === 'INFECTED') {
                logger.error(`Infected file found at ${keyName}`);
                throw new Error('Infected file suspected');
            } else {
                return avStatus;
            }
        });
}
```

Reply:
Good call, done